### PR TITLE
chore(test): upgrade Jest 29→30 and jsdom 20→29

### PR DESCRIPTION
**Pull request by tangentlin**
## Summary

Upgrades the Jest test stack to the latest versions while leaving Vitest untouched. Jest is the **only** jsdom consumer in the workspace (every Vitest package runs real-browser via Playwright, per ADR-013), so the jsdom bump is structurally isolated from Vitest.

Delivered in two stages:

**Stage A — Jest 29 → 30, jsdom 20 → 26**
- `jest`, `jest-environment-jsdom` → `^30` (resolves 30.4.2 / 30.4.1); `@jest/globals` → `^30` in the jest adapter; `@swc/jest` → `^0.2.39`. `pnpm dedupe` collapsed the leftover jest-29/jsdom-20 graph to a single jest 30.
- Two real Jest-30 breaking changes surfaced by running the suites (not by reading changelogs):
  - `expect().toThrowError()` was removed → migrated 2 call sites to `toThrow()` in `core`'s `dateUtil` test.
  - Stricter `roots` validation turned a missing `__tests__` dir into a hard error → removed the vestigial `test: jest` script from `component-driver-html` (it has zero tests — they live in `component-driver-html-test` — and no sibling driver package carries the script).
- Dropped the now-native **DOMRect** polyfill from the Radix setup (native in jsdom since 22.1; it was already inert on jsdom 26). Every other jsdom polyfill (ResizeObserver, matchMedia, Popover, dialog `showModal`, pointer-capture, `scrollIntoView`, `getBBox`, `structuredClone`, `TextEncoder`) stays — still unimplemented in jsdom 29.
- Aligned the `create-atomic-testing` scaffolder's emitted Jest versions to 30 (fixes a latent `@types/jest ^29.5.0` on a Jest-30 runtime for generated projects).

**Stage B — force latest jsdom 29**
- `jest-environment-jsdom@30` bundles jsdom 26; pinned it forward to the current-latest **jsdom 29.1.1** via a scoped `pnpm.overrides` entry (`jest-environment-jsdom>jsdom`) — no unsupported global override, no custom environment. Verified `jest-environment-jsdom@30` runs cleanly on jsdom 29.
- `engines.node` floor raised 22.12 → 22.13 (jsdom 29's requirement on the 22.x line; CI's Node-22-latest already satisfies it).

The jsdom-27 selector-engine swap (`nwsapi → @asamuzakjp/dom-selector`), the `click()→PointerEvent` change, and the jsdom-29 CSSOM overhaul do **not** affect the drivers: locators are accessibility/attribute-based (no selector-engine edge cases, no computed-style-string assertions), and clicks dispatch an explicit `FakeMouseEvent` rather than native `click()`.

**Validation:** all 19 Jest packages pass on both jsdom 26 and jsdom 29 with identical tallies (core 73, dom-core 45, react-legacy 2, create-atomic-testing 58, html 79, mui-v6 222, mui-v7 226, mui-v9 219+1 skip, mui-x ×4 = 92, vue-3 2, astryx 339, radix 146, primevue 56, fluent 161, conformance 18).

**Vitest:** untouched at runtime (browser mode never instantiates jsdom); only its unused optional-peer key floats 26 → 29 in the lockfile.

## Checks

- [x] `pnpm run check:type`
- [x] `pnpm run check:lint`
- [x] `pnpm run check:style`
- [x] `pnpm test:dom` (all 19 Jest packages, green on jsdom 26 and jsdom 29)
- [ ] `pnpm test:e2e` — not run locally (needs browsers + dev server); unaffected by the jsdom bump, exercised by CI

## Breaking change

- [ ] This PR introduces a breaking change (title uses `!`, e.g. `feat(core)!: ...`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01Qj4WWbWdj5TrGxLHSX7CAR)_

### CODE REVIEW

File: packages/create-atomic-testing/src/constants.ts [hunk 1]
```diff
@@ -64,11 +64,11 @@ export const THIRD_PARTY = {
...
-  jestEnvJsdom: { name: 'jest-environment-jsdom', range: '^29.7.0', dev: true },
-  swcJest: { name: '@swc/jest', range: '^0.2.37', dev: true },
+  jest: { name: 'jest', range: '^30.0.0', dev: true },
+  jestEnvJsdom: { name: 'jest-environment-jsdom', range: '^30.0.0', dev: true },
+  swcJest: { name: '@swc/jest', range: '^0.2.39', dev: true },
   swcCore: { name: '@swc/core', range: '^1.7.0', dev: true },
-  typesJest: { name: '@types/jest', range: '^29.5.0', dev: true },
+  typesJest: { name: '@types/jest', range: '^30.0.0', dev: true },
   vitest: { name: 'vitest', range: '^4.1.9', dev: true },
   jsdom: { name: 'jsdom', range: '^25.0.0', dev: true },
```
[the remark this answers is not in the document]
Comment (tangentlin) [reply packages/create-atomic-testing/src/constants.ts]: Bumped `THIRD_PARTY.jsdom` `^25.0.0` → `^29.0.0` in b61642c so generated Vitest(jsdom) setups match the repo's jsdom 29. REC-SYNC/coverage gates and the scaffolder unit tests stay green (this constant feeds only the vitest(jsdom) recipe and isn't version-cross-checked).

---
_Generated by [Claude Code](https://claude.ai/code)_
